backends/src/parsing.rs

```diff
@@ -398,22 +397,22 @@ impl<T: ParseLow> ParseHigh for ParseAdapter<T> {
                 }
             };
 
-            let storage = RefCell::new(self.0.storage_from_file(&file));
+            let storage = RefCell::new(backend.storage_from_file(&file));
 
             let walkable_functions = {
-                let mut local_functions = self.0.local_functions(&storage, &file)?;
+                let mut local_functions = backend.local_functions(&storage, &file)?;
                 local_functions.retain(|name, _| config.is_walkable_function(name));
                 local_functions
             };
 
-            let source_file = SourceFile::new(context.root.clone(), source_file.to_path_buf())?;
+            let source_file_obj = SourceFile::new(context.root.clone(), source_file.to_path_buf())?;
```

Comment:
```suggestion
            let source_file = SourceFile::new(context.root.clone(), source_file.to_path_buf())?;
```

This will necessitate some changes below, but I think those should be the last of the changes.

Could you please squash your commits down to one?